barretenberg/cpp/src/barretenberg/dsl/acir_format/ec_operations.cpp

```diff
@@ -13,19 +14,126 @@
 
 namespace acir_format {
 
+// This functions assumes that:
+// 1. the points are on the curve
```

Comment:
What if the points are not on the curve? Where do we check that?

Reply:
I added a check for this case:
- if x match and y does not match, at compile time, then return 0
- else, if x match, then return error
- else: we assume x-coordinates are distincts and will do the unconditional add